launch.go

```diff
@@ -213,32 +233,43 @@ func launch(api screwdriver.API, buildID string, rootDir string) error {
 }
 
 // Executes the command based on arguments from the CLI
-func launchAction(c *cli.Context) error {
-	url := c.String("api-uri")
-	token := c.String("token")
-	workspace := c.String("workspace")
-	buildID := c.Args().Get(0)
-
-	if buildID == "" {
-		return cli.ShowAppHelp(c)
-	}
-
+func launchAction(api screwdriver.API, buildID string, rootDir string) error {
 	log.Printf("Starting Build %v\n", buildID)
 
-	api, err := screwdriver.New(url, token)
-	if err != nil {
-		log.Fatalf("Error creating Screwdriver API %v: %v", buildID, err)
-	}
+	if err := launch(api, buildID, rootDir); err != nil {
+		if _, ok := err.(executor.ErrStatus); ok {
+			log.Printf("Failure due to non-zero exit code: %v\n", err)
+		} else {
+			log.Printf("Error running launcher: %v\n", err)
+		}
 
-	if err = launch(api, buildID, workspace); err != nil {
-		log.Fatalf("Error running launcher: %v\n", err)
-		os.Exit(1)
+		exit(screwdriver.Failure, api)
+		return nil
 	}
 
+	exit(screwdriver.Success, api)
 	return nil
 }
 
+func recoverPanic(api screwdriver.API) {
+	if p := recover(); p != nil {
+		filename := fmt.Sprintf("launcher-stacktrace-%s", time.Now().Format(time.RFC3339))
+		tracefile := filepath.Join(os.TempDir(), filename)
+
+		log.Printf("ERROR: Internal Screwdriver error. Please file a bug about this: %v", p)
+		log.Printf("ERROR: Writing StackTrace to %s", tracefile)
+		err := ioutil.WriteFile(tracefile, debug.Stack(), 0600)
```

Comment:
Ok, I just pushed a change that will try one last time to output something useful before dying: finalRecover()